scripts/regression/test_runner.py

```diff
@@ -151,32 +161,70 @@ class BenchmarkResult:
+            try:
+                improved_by = int((float(old_res) - float(new_res)) * 100.0 / float(old_res))
+            except Exception:
+                improved_by = 0
+            if improved_by > 0:
+                improvements_list.append(BenchmarkResult(benchmark, old_res, new_res, improvement=improved_by))
+            else:
+                other_results.append(BenchmarkResult(benchmark, old_res, new_res))
     benchmark_list = [res.benchmark for res in regression_list]
 
+time_old = geomean(old_runner.complete_timings)
+time_new = geomean(new_runner.complete_timings)
+
 exit_code = 0
 regression_list.extend(error_list)
 summary = []
 if len(regression_list) > 0:
-    exit_code = 1
-    print(
-        '''====================================================
-==============  REGRESSIONS DETECTED   =============
+    if not (isinstance(time_old, str) or isinstance(time_new, str)):
+        regression_percentage = int((time_new - time_old) * 100.0 / time_old)
+    else:
+        regression_percentage = 0
+    if (
+        isinstance(MAX_ALLOWED_REGRESS_PERCENTAGE, int)
+        and isinstance(regression_percentage, int)
+        and regression_percentage < MAX_ALLOWED_REGRESS_PERCENTAGE
+    ):
+        # allow individual regressions less than 10% when overall geomean had improved or hadn't change (on large benchmarks)
+        regressions_header = 'ALLOWED REGRESSIONS'
+    else:
+        regressions_header = 'REGRESSIONS DETECTED'
+        exit_code = 1
+
+    if len(regression_list) > 0:
+        print(
+            f'''====================================================
+==============  {regressions_header}   ==============
 ====================================================
 '''
-    )
-    for regression in regression_list:
-        print(f"{regression.benchmark}")
-        print(f"Old timing: {regression.old_result}")
-        print(f"New timing: {regression.new_result}")
-        if regression.old_failure or regression.new_failure:
-            new_data = {
-                "benchmark": regression.benchmark,
-                "old_failure": regression.old_failure,
-                "new_failure": regression.new_failure,
-            }
-            summary.append(new_data)
-        print("")
+        )
+
+        for regression in regression_list:
+            print("")
+            print(f"{regression.benchmark}")
+            print(f"Old timing: {regression.old_result}")
+            print(f"New timing: {regression.new_result}")
+            # add to the FAILURES SUMMARY
+            if regression.old_failure or regression.new_failure:
+                new_data = {
+                    "benchmark": regression.benchmark,
+                    "old_failure": regression.old_failure,
+                    "new_failure": regression.new_failure,
+                }
+                summary.append(new_data)
+            print("")
+
+        if not (isinstance(time_old, str) or isinstance(time_new, str)):
```

Comment:
will these values every be a string? If they are assigned on the lines
```
time_old = geomean(old_runner.complete_timings)
time_new = geomean(new_runner.complete_timings)
```
They should be doubles right? or ints?